### PR TITLE
Steven reccommended fixes

### PR DESCRIPTION
**Pull request by liammccartney**
@zahidkarim - implements most of steven's reccommended fixes, rms unneeded code and files, and fixes double eventing

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -16,4 +16,10 @@
 //= require turbolinks
 //= require_tree .
 
-$(function(){ $(document).foundation(); });
```
Comment (sgharms) [inline app/assets/javascripts/application.js]: @liammccartney Nope.  Don't put code in here.  How about an `initialize.js` file.  The closure pattern here (known as an IIFE) is nice though.

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -16,4 +16,10 @@
...
+$(function(){
+  $(document).foundation();
+  SC.initialize({
+    client_id: "b99714ff12392f9d7e1498ab5166732d",
```
Comment (sgharms) [inline app/assets/javascripts/application.js]: When you extract the above (heh-heh) to `initialize.js.erb`, you could pass this key in from `ENV`.  That would be sick yo.

File: app/assets/javascripts/soundcloud_stream.js.erb [hunk 3]
```diff
@@ -1,50 +1,42 @@
...
-    client_id: "<%= ENV['SOUNDCLOUD_APP_ID'] %>",
-    redirect_uri: 'http://cryptic-dusk-7305.herokuapp.com/callback'
-  });
-
-  $('#connect').click(function(){
-    var currentUser = new SoundcloudUser
-    currentUser.searchSongs();
-  })
+$('#connect').click(function(){
+  new SoundcloudUser
```
Comment (sgharms) [inline app/assets/javascripts/soundcloud_stream.js.erb]: it's more extensible-r to do `SoundcouldUser()`

File: app/assets/javascripts/soundcloud_stream.js.erb [hunk 4]
```diff
@@ -1,50 +1,42 @@
...
+})
 
-  $('.track').on('click', function(event){
-    track = new Track({id: event.currentTarget.dataset.trackId, source_id: event.currentTarget.dataset.sourceId})
-    track.audio.play()
-  })
+$('.user-tracks').on('click', '.track', function(event){
+  track = new Track({id: event.currentTarget.dataset.trackId, source_id: event.currentTarget.dataset.sourceId})
```
Comment (sgharms) [inline app/assets/javascripts/soundcloud_stream.js.erb]: most people write it like


```javascript
someFunction({
 key: val,
 key2: val2
});

WIDE lines are hard to read (think about newspaper column layout)

File: app/assets/javascripts/soundcloud_stream.js.erb [hunk 5]
```diff
@@ -1,50 +1,42 @@
...
-  $('.snippet').on('click', function(event){
-    var snipData = event.currentTarget.dataset;
-    var snippetTrack = new Track({source_id: snipData.sourceId});
-    snippet = new Snippet({track: snippetTrack, startTime: snipData.startTime, endTime: snipData.endTime
-    });
-    snippet.playSnip();
-  })
+$('.user-snippets').on('click', '.snippet', function(event){
+  var snipData = event.currentTarget.dataset;
+  var snippetTrack = new Track({source_id: snipData.sourceId});
```
Comment (sgharms) [inline app/assets/javascripts/soundcloud_stream.js.erb]: Again, don't take my previous comment as an excuse to reformat everything (like here) .... just don't go crazy with it.

File: app/assets/javascripts/soundcloud_stream.js.erb [hunk 6]
```diff
@@ -1,50 +1,42 @@
...
-    $.ajax({
-      url: '/snippets',
-      method: 'POST',
-      data: {snippet: {track_id: newSnippet.track_id, start_time: newSnippet.start_time, end_time: newSnippet.end_time}},
-    }).done(function(response){
-      $('.user-snippets').append(response);
-    })
+$('.columns#right').on('click', '#snip', function(event){
+  var endTime = track.audio.currentTime;
+  var startTime = track.audio.currentTime - 15;
```
Comment (sgharms) [inline app/assets/javascripts/soundcloud_stream.js.erb]: `15` is a "MAGIC NUMBER".  Why 15?  Why not 17.5 or 3.14?  A classic refactoring is

"REPLACE MAGIC NUMBER WITH SYMBOLIC CONSTANT"

So like

`DEFAULT_SAMPLE_SIZE = 15` 

and then 

` track.audio.currentTime - DEFAULT_SAMPLE_SIZE`

that tells me much more.  think 'bout it.

**Comment by sgharms** [comment]
@zahidkarim @torpeyp @liammccartney @chrisrcoles 

I made some "pro tip" annotations in this commit.  These are some subtle refactorings that communicate a place you will want to go in terms of making your code _expressive_.  This is what we all aim for in this biz and speaks well for you in interview.

**Comment by liammccartney** [comment]
@sgharms thank you!

**Pull request merged**